helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -92,19 +99,51 @@ public ZkBaseDataAccessor(HelixZkClient zkClient) {
     _zkClient = zkClient;
   }
 
+  /**
+   * Lazy initialization is used to reduce the cost of unnecessary opened zkConnections
+   * @return singleton instance of non-ZnRecord ZkClient
+   */
+  private HelixZkClient getNonZNRecordZkClient() {
+    if (_nonZNRecordZkClient == null) {
+      synchronized (ZkBaseDataAccessor.class) {
+        _nonZNRecordZkClient = new ZkClient.Builder().setConnection(
```

Comment:
We may need to do a null check again here to avoid multiple runs of the creation logic by other threads that were blocking.

Reply:
Good catch!